extra/mariabackup/xtrabackup.cc

```diff
@@ -5106,6 +5157,14 @@ class BackupStages {
 				return false;
 			}
 
+			// SHOW ENGINE INNODB STATUS would be written
+			// to mysql.general_log, which could be an
+			// ENGINE=CSV table that we just copied
+			// above.
+			if (!backup_wait_for_commit_lsn()) {
+				return false;
+			}
+
```

Comment:
Yes, this is trying to address the point that @montywi raised: backup is apparently waiting for too much log to be copied. What users would observe was the message before `backup_wait_for_lsn(server_lsn_after_lock)` and nothing about the commit LSN that still needs to be reached. I think that I must revise this to include the messages.

Since we are blocking transaction commits, any currently active transactions would have to be rolled back after the backup is restored. In fact, copying less log could be better, because less of those incomplete transactions would be copied.

Reply:
Thanks, that makes perfect sense with transactions being blocked only at commit. If there are long running transactions, it would cause a lot of unnecessary roll forward and rollback.

Reply:
5458b8c85dc7073bc0caa8859d5a81d087199ee9 clarifies this both in a source code comment and in the commit message.